sql/moz-fx-data-shared-prod/firefox_ios_derived/firefox_ios_clients_v1/query.sql

```diff
@@ -0,0 +1,285 @@
+        DATE(submission_timestamp) = @submission_date
+        AND client_info.client_id IS NOT NULL
+        AND ping_info.seq = 0 -- Pings are sent in sequence, this guarantees that the first one is returned.
+    )
+  GROUP BY
+    client_id
+),
+-- Find earliest data per client from the metrics ping.
+metrics_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    DATETIME(
+      MIN(submission_timestamp)
+    ) AS min_submission_datetime,  -- do we need this? Looks like later we might want to see timestamp corresponding to the row that contains the campaign info
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+    -- # TODO: understand better the install_source field.
+    -- what exactly does it contain?
+    ARRAY_AGG(install_source IGNORE NULLS ORDER BY submission_timestamp ASC)[
+      SAFE_OFFSET(0)
+    ] AS install_source,
+  FROM
+    (
+      SELECT
+        client_info.client_id AS client_id,
+        sample_id,
+        submission_timestamp,
+        NULLIF(metrics.string.metrics_adjust_ad_group, "") AS adjust_ad_group,
+        NULLIF(metrics.string.metrics_adjust_campaign, "") AS adjust_campaign,
+        NULLIF(metrics.string.metrics_adjust_creative, "") AS adjust_creative,
+        NULLIF(metrics.string.metrics_adjust_network, "") AS adjust_network,
+        NULLIF(metrics.string.metrics_install_source, "") AS install_source,
+      FROM
+        `org_mozilla_firefox.metrics`
+      WHERE
+        DATE(submission_timestamp) = @submission_date
+        AND client_info.client_id IS NOT NULL
+    )
+  GROUP BY
+    client_id
+),
+_current AS (
+  SELECT
+    * EXCEPT (min_submission_datetime, first_run_date, sample_id, adjust_info),
+    COALESCE(first_session.adjust_info, metrics.adjust_info) AS adjust_info,
+    COALESCE(first_seen.sample_id, first_session.sample_id, metrics.sample_id) AS sample_id,
+    -- activated AS activated,  -- # TODO: new_profile_activation needs to be created before we can get this
+    STRUCT(
+      IF(first_session.client_id IS NULL, FALSE, TRUE) AS reported_first_session_ping,
+      IF(metrics.client_id IS NULL, FALSE, TRUE) AS reported_metrics_ping,
+      DATE(first_session.min_submission_datetime) AS min_first_session_ping_submission_date,
+      DATE(first_session.first_run_date) AS min_first_session_ping_run_date,
+      DATE(
+        metrics.min_submission_datetime
+      ) AS min_metrics_ping_submission_date, -- TODO: are these date fields useful?
+      -- # TODO: not quite sure what this case statement is meant to do?
```

Comment:
It's trying to tell us where attribution came from: the first_session ping or the metrics ping. Ignore it, and focus on getting the logic right for your change.

Reply:
But isn't that just the same as this then?
https://github.com/mozilla/bigquery-etl/pull/3709/files#diff-30f0504d5c94b0a17cc26dc05c573361aa9442c615f0dbd37ad713d9815ce5f8R175-R181

Reply:
I believe that way is correct as well. I've never used this field though so I'm not sure that it's important

Reply:
I think at some point we were discussing if we wanted to choose whichever ping came first, or if we wanted to default to first_session ping, which I don't have a preference since there are very few occurrences where they don't match and there are no right answers for those cases.

Reply:
For now I commented out those fields. If we agree we do not need them then I can remove the commented out code before merging.